src/gc.c

```diff
@@ -646,7 +646,7 @@ static int mark_reset_age = 0;
 
 static int64_t scanned_bytes; // young bytes scanned while marking
 static int64_t perm_scanned_bytes; // old bytes scanned while marking
-static int prev_sweep_full = 1;
+int prev_sweep_full = 1;
```

Comment:
it seems that this is always set to 1? i.e. set to 1 here and never changed anywhere else?

Reply:
no i think it's set, here:
https://github.com/vilterp/julia/blob/2b2c3ab4db8c35f4deb8a44330dc765e73e61647/src/gc.c#L3261

Jameson suggested this change. the idea is to make sure that we're only doing the GC Mark edge-reporting during the _second_ GC, since our call to `jl_gc_collect(JL_GC_FULL)` does two mark+sweeps: 1. mark, 2. sweep FULL, 3. mark, 4. sweep INCREMENTAL. Step 3 will be a full mark, because all mark bits have been reset. :)

Reply:
oh I see, this variable was already defined here (and set elsewhere, before this commit), but you added an extern declaration to heap-snapshot.h so we could check it there… gotcha. wasn't reading the diff close enough. nice change!